docs/api-guide.md

```diff
@@ -697,17 +689,28 @@ Put put =
         .build();
 ```
 
-#### `Delete` operation
+##### Blind `Put` operation
```

Comment:
In my current understanding, this new feature (blind write) seems to work as follows.

1. If `record A` exists and we run `put(record A)` **without** `blind()`, ScalarDB automatically reads `record A` to create the read set, and `put()` (i.e., UPDATE) will succeed.
2. If `record A` exists and we run `put(record A)` **with** `blind()`, ScalarDB returns a conflict error since there is no `record A` information on the read set.
   * This is the same behavior as current (v3) ScalarDB (i.e., same as `record A` exists and we run `put(record A)` without any `get()`/`scan()`, in current (v3) ScalarDB).
3. If `record A` does NOT exist and we run `put(record A)` **with** `blind()`, ScalarDB puts a new record without read operation (there is a performance benefit).
   * This is the same behavior as current (v3) ScalarDB (i.e., same as `record A` does NOT exist and we run `put(record A)`, in current (v3) ScalarDB).
4. If `record A` does NOT exist and we run `put(record A)` **without** `blind()`, ...

---

### Question 1

Regarding `1.`, `2.`, and `3.`, is my understanding correct?

---

### Question 2

What will happen in `4.`?
If it puts (INSERT) `record A` as a new record, what is the difference between `3.` and `4.`? (Does ScalarDB try to read the null/empty record in case of `4.`?)

Reply:
Note that, after some discussion, the `blind()` method is renamed to `disableImplicitPreRead()`.

For question 1, yes, your understanding is correct for `1.`, `2.`, and `3.`.

For question 2, in case of `4.`, ScalarDB tries to read record A, but it doesn't exist. So ScalarDB tries to put a new record without information from any existing record, which is the same behavior of `3.`. Therefore, the difference between `3.` and `4.` is whether or not ScalarDB tries to read the record.

Reply:
I see. I understand!
Thank you for your explanation!